gateway/ln-gateway/src/lightning/ldk.rs

```diff
@@ -0,0 +1,501 @@
+                    short_channel_id: None,
+                    incoming_chan_id: 0,
+                    htlc_id: 0,
+                }))
+                .await?;
+        }
+
+        // The `PaymentClaimable` event is the only event type that we are interested
+        // in. We can safely ignore all other events.
+        node.event_handled();
+
+        Ok(())
+    }
+
+    /// Converts a transaction outpoint to a short channel ID by querying the
+    /// blockchain.
+    async fn outpoint_to_scid(&self, funding_txo: OutPoint) -> anyhow::Result<u64> {
+        let block_height = self
+            .esplora_client
+            .get_merkle_proof(&funding_txo.txid)
+            .await?
+            .ok_or(anyhow::anyhow!("Failed to get merkle proof"))?
+            .block_height;
+
+        let block_hash = self.esplora_client.get_block_hash(block_height).await?;
+
+        let block = self
+            .esplora_client
+            .get_block_by_hash(&block_hash)
+            .await?
+            .ok_or(anyhow::anyhow!("Failed to get block"))?;
+
+        let tx_index = block
+            .txdata
+            .iter()
+            .enumerate()
+            .find(|(_, tx)| tx.txid() == funding_txo.txid)
+            .ok_or(anyhow::anyhow!("Failed to find transaction"))?
+            .0 as u32;
+
+        let output_index = funding_txo.vout;
+
+        Ok(create_short_channel_id(
+            block_height,
+            tx_index,
+            output_index,
+        ))
+    }
+}
+
+impl Drop for GatewayLdkClient {
+    fn drop(&mut self) {
+        self.htlc_stream_seeder_task_handle.abort();
+
+        info!("Stopping LDK Node...");
+        if let Err(err) = self.node.stop() {
+            error!("Failed to stop LDK Node: {err:?}");
+        } else {
+            info!("LDK Node stopped.");
+        }
+    }
+}
+
+#[async_trait]
+impl ILnRpcClient for GatewayLdkClient {
+    async fn info(&self) -> Result<GetNodeInfoResponse, LightningRpcError> {
+        let node_status = self.node.status();
+
+        let chain_tip_block_summary = self
+            .esplora_client
+            .get_blocks(None)
+            .await
+            .unwrap()
+            .into_iter()
+            .next()
+            .unwrap();
+
+        let esplora_chain_tip_timestamp = chain_tip_block_summary.time.timestamp;
+        let block_height: u32 = chain_tip_block_summary.time.height;
+
+        let synced_to_chain = node_status.latest_wallet_sync_timestamp.unwrap_or_default()
+            > esplora_chain_tip_timestamp
+            && node_status
+                .latest_onchain_wallet_sync_timestamp
+                .unwrap_or_default()
+                > esplora_chain_tip_timestamp;
+
+        Ok(GetNodeInfoResponse {
+            pub_key: self.node.node_id().serialize().to_vec(),
+            // TODO: This is a placeholder. We need to get the actual alias from the LDK node.
+            alias: "LDK Fedimint Gateway Node".to_string(),
```

Comment:
does ldk node not expose this yet?

Reply:
Not as far as I can see

Reply:
Perhaps add a random identifier in the alias here.

Reply:
Done. I just appended the node's pubkey.